core/src/Kokkos_HostSpace.hpp

```diff
@@ -165,6 +166,15 @@ class HostSpace {
                   const size_t arg_alloc_size,
                   const size_t arg_logical_size = 0) const;
 
+  void* impl_allocate(const char* arg_label, const size_t arg_alloc_size,
```

Comment:
Why does it need to be public?  Where will it be used?

Reply:
@dalg24 and I are discussing on Slack, but I'll record it here as well. It need to be public because this is infrastructure to enable LogicalMemorySpaces, which are separate classes in which this guy will be used. We're now discussing whether to make this public or use a friend declaration

Reply:
What we determined (though Slack and meetings and the like) is that we'll make this private for the moment, and figure out what more to do later (though I'm leaning towards friending the LogicalMemorySpace)